packages/browser-utils/src/networkUtils.ts

```diff
@@ -54,3 +54,64 @@ export function getFetchRequestArgBody(fetchArgs: unknown[] = []): RequestInit['
 
   return (fetchArgs[1] as RequestInit).body;
 }
+
+/**
+ * Parses XMLHttpRequest response headers into a Record.
+ * Extracted from replay internals to be reusable.
+ */
+export function parseXhrResponseHeaders(xhr: XMLHttpRequest): Record<string, string> {
+  let headers: string | undefined;
+  try {
+    headers = xhr.getAllResponseHeaders();
+  } catch (error) {
+    DEBUG_BUILD && debug.error(error, 'Failed to get xhr response headers', xhr);
+    return {};
+  }
+
+  if (!headers) {
+    return {};
+  }
+
+  return headers.split('\r\n').reduce((acc: Record<string, string>, line: string) => {
+    const [key, value] = line.split(': ') as [string, string | undefined];
+    if (value) {
+      acc[key.toLowerCase()] = value;
+    }
+    return acc;
+  }, {});
+}
+
+/**
+ * Gets specific headers from a Headers object (Fetch API).
+ * Extracted from replay internals to be reusable.
+ */
+export function getFetchResponseHeaders(headers: Headers, allowedHeaders: string[]): Record<string, string> {
```

Comment:
Q: I don't see this being used outside of replay. What's the reason for exporting it from browser-utils? Do we need it for follow-up work?
(I might be missing something though!)

If we don't need it in multiple packages, let's keep it in replay for the time being to this PR atomic.